### PR TITLE
Fix all eslint issues

### PR DESCRIPTION
**Pull request by c-w**
This pull request fixes all eslint issues and sets up CI so that we always run tests and lint on every check-in/pull-request.

The pull request also fixes a few bugs uncovered by eslint, mostly to do with referencing undefined variables.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+language:
+  node_js
+
+node_js:
+  - "7"
```
Comment (Smarker) [inline .travis.yml:5]: package.json has node version 6.0.0 not sure if we want to keep our version at 7 or change what is in .travis.yml from 7 to 6.

**Review by Smarker — APPROVED** [review]
Looks good. I see that I will have to make changes in my code since I might have used double quotes a couple of times rather than single quotes.

File: .travis.yml (hunk 1 above)
Comment (c-w) [reply .travis.yml:5]: Good catch. We can run on both actually, see 0aa6524.

Comment (c-w) [reply .travis.yml:5]: NB: you can use `./node_modules/.bin/eslint src --fix` to auto-fix simple issues like double-quotes vs single-quotes.

**Review by erikschlegel — COMMENTED** [review]
Thank you!

**Pull request merged**